### DISCUSSION TITLE
Need a way to remove the need to build external sub packages when developing

### DISCUSSION THREAD

**Discussion by MartinDawson** [vercel/turborepo#6583] [Help]
These 2 links describe nice solutions:

https://turbo.build/blog/you-might-not-need-typescript-project-references
https://turbo.build/repo/docs/handbook/sharing-code/internal-packages

However, these are for internal packages.

We need an equivalent for external packages. Currently the dev experience sucks for them. Every change, you have to stop your dev server and rebuild the package.

Even having `dev: tsc --watch` in the package json scripts isn't ideal because this still requires the project to build when developing on each change and doesn't work properly when you have child packages that need building first.

Their needs to be some type of feature where we can do this:

` "main": "./src/index.ts"`

But then auto convert this to the dist js path (i.e "main": "./build/esm/index.js" when the project is built on CI/CD and published to the app.

**Comment by MartinDawson** [comment]
Any Ideas?

**Comment by mehulkar** [comment]
@MartinDawson are you thinking this is relevant for turborepo or turbopack? This kind of thing depends on the frameworks and build tools you're using. It's currently outside the scope of Turborepo, but if you're in Next.js, it may fall into Turbopack's territory

**Reply by MartinDawson** [reply]
Thanks for the reply.

Turborepo. This is a very common issue and turborepo falls short when you have external packages.

I haven't actually used turbopack at all. We use vite.

**Reply by mehulkar** [reply]
Could you provide an example? At first I thought you were developing an external package and symlinking into your monorepo's `node_modules`? But in your original post you said:

> when the project is built on CI/CD and published to the app.

So I am confused about what you actually mean and when and why you need to rebuild external dependencies

**Comment by jswhisperer** [comment]
https://nextjs.org/docs/app/api-reference/config/next-config-js/serverExternalPackages

**Comment by anthonyshew** [comment]
I'm going to close this one due to age and lack of clarity. If an example gets provided for the original post, we can re-assess!
